lua/conform/formatters/deno_fmt.lua

```diff
@@ -16,11 +34,27 @@ return {
   },
   command = "deno",
   args = function(self, ctx)
+    local extension = extensions[vim.bo[ctx.buf].filetype]
+
+    if
+      vim.tbl_get(unstable_extensions, extension)
+      and not vim.list_contains(
+        vim.tbl_get(conform.formatters, "deno_fmt", "append_args") or {},
+        "--unstable-component"
+      )
+    then
+      log.warn(
+        "You are trying to format an unstable file type (."
+          .. extension
+          .. ") without the corresponding `--unstable-component` flag. Add the flag to `append_args` to format your code. See the Deno documentation for more information: https://docs.deno.com/runtime/reference/cli/formatter/#formatting-options-unstable-component"
+      )
+    end
```

Comment:
Could we just add `--unstable-component` automatically for the user here? If the formatter is running then that means they must have explicitly encoded _somewhere_ that they want to use deno fmt for that filetype. Are there any surprising risks to adding that flag?

Reply:
That's a good point!

I don't think there are any risks in adding the flag. Even if the flag causes some problems, the formatting is the only thing that will suffer. The user will then know that `deno_fmt` causes the problem since they've chosen it as the formatter for the file they are trying to format. If they look it up, they will most likely come across this unstable flag, which explains by its name that everything might not be working 100 % yet.

The situation would be different if you could format the unstable files both with and without the flag, but in the current situation, you must include that flag. Otherwise, `deno fmt` succeeds, but no formatting takes place.

This would also fix this TODO:

> I'm unsure how to check for the `"unstable": ["fmt-component]` config option in `deno.json`, so that's still missing.

---

I could then change the MR so that the if statement adds `--unstable-component` to the return value and log it to `log.info` for debugging purposes. How does that sound?